test/test_python_dispatch.py

```diff
@@ -551,6 +551,24 @@ def foo_impl(x, y, z, w):
             getattr(torch.ops, self.test_ns).foo.default,
             getattr(torch.ops, self.test_ns).foo_functional.default, (x, y, z, w))
 
+    def test_register_fallthrough(self):
+        my_lib = Library('aten', 'IMPL')
+        my_lib.impl("mm", torch.library.Fallthrough(), "AutocastCPU")
+
+        a = torch.randn(2, 3, device='cpu', dtype=torch.float32)
+        b = torch.randn(3, 2, device='cpu', dtype=torch.float32)
+        with torch.autocast(device_type="cpu", dtype=torch.bfloat16):
+            # dtype for mm should be float32 since we registered a fallthrough
+            self.assertTrue(torch.mm(a, b).dtype == torch.float32)
+            # ops that don't have a fallthrough registered should not be affected
+            self.assertTrue(torch.matmul(a, b).dtype == torch.bfloat16)
+
+        del my_lib
```

Comment:
Should add a try...finally so that if the test fails, we don't keep the changes to PyTorch permanently

Reply:
if an error gets raised, the stack trace holds onto my_lib :/

Reply:
And the test framework actually keeps it alive while it's running the other tests? :o

Reply:
Yes :P

Reply:
This used to happen a lot in the custom ops test, one test failure would cause a bunch of other tests to fail (because they all tried to register the same operator "custom::foo" and relied on the library going out of scope after each test, but after a failure the library did not actually get deleted)

Reply:
Ho wow this is very sad... Ok!